src/query_executor.c

```diff
@@ -162,13 +169,18 @@ void inlineProperties(AST_Query *ast) {
             SIValue *key;
             SIValue *val;
 
+            // Build the left-hand filter value from the node alias and property
             Vector_Get(properties, j, &key);
-            Vector_Get(properties, j+1, &val);
+            property = key->stringval;
+            lhs = New_AST_AR_EXP_VariableOperandNode(alias, property);
 
-            const char *alias = entity->alias;
-            const char *property = key->stringval;
+            // Build the right-hand filter value from the specified constant
+            // TODO can update grammar so that this constant is already an ExpressionNode
+            // instead of an SIValue
```

Comment:
I think it would be better to leave this modification for a separate PR. Inlined filters are handled by the same rules as property assignment in node/edge creation, so it's a fairly large area to modify!